daphne/src/lib.rs

```diff
@@ -868,6 +868,10 @@ impl<S> DapRequest<S> {
             Err(DapAbort::UnrecognizedMessage)
         }
     }
+
+    pub(crate) fn host(&self) -> String {
```

Comment:
Do we expect the hostname to not exist under normal conditions? If not failing close might be more appropriate (i.e., return an error here instead of silently defaulting to the empty string). Otherwise, I'm not sure defaulting to the empty string is appropriate. Maybe something like "UNKNOWN_HOST"?
```suggestion
    pub(crate) fn host(&self) -> Result<String, DapAbort> {
```